pkg/skaffold/initializer/init.go

```diff
@@ -150,94 +172,136 @@ func DoInit(out io.Writer, c Config) error {
 	return nil
 }
 
-func processCliArtifacts(artifacts []string) ([]dockerfilePair, error) {
-	var pairs []dockerfilePair
+// autoSelectBuilders takes a list of builders and images, checks if any of the builders' configured target
+// images match an image in the image list, and returns a list of the matching builder/image pairs. Also
+// returns the images from the original image list that didn't match any build configurations.
+func autoSelectBuilders(buildConfigs []InitBuilder, images []string) ([]BuilderImagePair, []string) {
```

Comment:
Note: This function is only really useful for builders that can configure a target image themselves, so there will be a test for this function in the Jib follow-up PR, since Jib makes better use of this than Dockerfiles.